### PR TITLE
Instrument Aerospike V3

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
Adds support for aerospike v3.16.2 - v3.16.7

### Motivation
feature request

### Additional Notes
Only support for v3.16.2 - v3.16.7 is added because those are the only versions that support Node.js 14. The rest of the semver minors only have support for Node.js versions < 14

### CODE REVIEW

**Comment by tlhunter** [comment]
Hmm... I suppose this is a little hard to test since the PR isn't running with Node.js v14. The CI job is skipped:
https://github.com/DataDog/dd-trace-js/actions/runs/7037932351/job/19153797867?pr=3830

I'm guessing you ran it locally and things passed?

**Comment by khanayan123** [comment]
Test output when ran with a tracer major version of 3 on node 14:

image

**Comment by khanayan123** [comment]
Test output:

`Run yarn test:plugins:ci
yarn run v1.22.21
$ yarn services && nyc --no-clean --include "packages/datadog-instrumentations/src/@($(echo $PLUGINS)).js" --include "packages/datadog-instrumentations/src/@($(echo $PLUGINS))/**/*.js" --include "packages/datadog-plugin-@($(echo $PLUGINS))/src/**/*.js" -- npm run test:plugins
$ node ./scripts/install_plugin_modules && node packages/dd-trace/test/setup/services
$ yarn --ignore-engines
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1701285483088-0.07316008757948311/node but npm is using /__t/node/[14](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:15).21.3/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> dd-trace@3.[15](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:16).1 test:plugins /__w/dd-trace-js/dd-trace-js
> mocha --colors --exit -r "packages/dd-trace/test/setup/mocha.js" "packages/datadog-instrumentations/test/@($(echo $PLUGINS)).spec.js" "packages/datadog-plugin-@($(echo $PLUGINS))/test/**/*.spec.js"

Warning: Cannot find any files matching pattern "packages/datadog-instrumentations/test/@(aerospike).spec.js"


  Plugin
    aerospike
      with aerospike 3.[16](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:17).2 - 3.16.7 (3.16.2)
        without configuration
          client
Nov 29 2023 19:[18](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:19):46 GMT: WARN (7335) [as_pipe.c:208] [read_file] - Failed to open /proc/sys/net/core/wmem_max for reading
Nov 29 2023 [19](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:20):18:46 GMT: WARN (7335) [as_pipe.c:267] [get_buffer_size] - Failed to read /proc/sys/net/core/wmem_max; should be at least 5242880. Please verify.
Nov 29 [20](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:21)[23](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:24) 19:18:46 GMT: WARN (7335) [as_pipe.c:208] [read_file] - Failed to open /proc/sys/net/core/rmem_max for reading
Nov 29 2023 19:18:46 GMT: WARN (7335) [as_pipe.c:267] [get_buffer_size] - Failed to read /proc/sys/net/core/rmem_max; should be at least 15728640. Please verify.
            ✓ should instrument put
            ✓ should instrument connect
            ✓ should instrument get
            ✓ should instrument operate
            ✓ should instrument createIndex
            ✓ should run the callback in the parent context
            ✓ should handle errors
            peer service computation
              ✓ should compute peer service
            service and operation naming
              in version v0
                ✓ should conform to the naming schema
              in version v1
                ✓ should conform to the naming schema
              service naming short-circuit in v0
                ✓ should pass service name through
        with configuration
          ✓ should be configured with the correct values
          service and operation naming
            in version v0
              ✓ should conform to the naming schema
            in version v1
              ✓ should conform to the naming schema
            service naming short-circuit in v0
              ✓ should pass service name through
      with aerospike 3.16.2 - 3.16.7 (3.16.7)
        without configuration
          client
Nov 29 2023 19:18:46 GMT: WARN (7335) [as_pipe.c:208] [read_file] - Failed to open /proc/sys/net/core/wmem_max for reading
Nov 29 2023 19:18:46 GMT: WARN (7335) [as_pipe.c:267] [get_buffer_size] - Failed to read /proc/sys/net/core/wmem_max; should be at least 5[24](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:25)2880. Please verify.
Nov 29 2023 19:18:46 GMT: WARN (7335) [as_pipe.c:208] [read_file] - Failed to open /proc/sys/net/core/rmem_max for reading
Nov 29 2023 19:18:46 GMT: WARN (7335) [as_pipe.c:[26](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:27)7] [get_buffer_size] - Failed to read /proc/sys/net/core/rmem_max; should be at least 157[28](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:29)640. Please verify.
            ✓ should instrument put
            ✓ should instrument connect
            ✓ should instrument get
            ✓ should instrument operate
            ✓ should instrument createIndex
            ✓ should run the callback in the parent context
            ✓ should handle errors
            peer service computation
              ✓ should compute peer service
            service and operation naming
              in version v0
                ✓ should conform to the naming schema
              in version v1
                ✓ should conform to the naming schema
              service naming short-circuit in v0
                ✓ should pass service name through
        with configuration
          ✓ should be configured with the correct values
          service and operation naming
            in version v0
              ✓ should conform to the naming schema
            in version v1
              ✓ should conform to the naming schema
            service naming short-circuit in v0
              ✓ should pass service name through

Slack Reporter is disabled

  [30](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:31) passing (483ms)

------------------------------|---------|----------|---------|---------|---------------------
File                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s   
------------------------------|---------|----------|---------|---------|---------------------
All files                     |   86.15 |    70.83 |   94.73 |   90.[32](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:33) |                     
 datadog-instrumentations/src |      85 |       50 |     100 |     100 |                     
  aerospike.js                |      85 |       50 |     100 |     100 | 12-28               
 datadog-plugin-aerospike/src |   86.66 |    77.77 |   92.85 |   86.66 |                     
  index.js                    |   86.66 |    77.77 |   92.85 |   86.66 | 54-55,74-75,102-103 
------------------------------|---------|----------|---------|---------|---------------------
Done in [43](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:44).[60](https://github.com/DataDog/dd-trace-js/actions/runs/7037442343/job/19152225822#step:11:61)s.`

**Review by tlhunter — APPROVED** [review]

File: packages/datadog-instrumentations/src/aerospike.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function wrapProcess (process) {
 addHook({
   name: 'aerospike',
   file: 'lib/commands/command.js',
-  versions: ['4', '5']
+  versions: ['^3.16.2', '4', '5']
```
Comment (rochdev) [inline packages/datadog-instrumentations/src/aerospike.js:43]: Didn't notice this before, but this caps the version to v5, so when v6 is released the instrumentation will stop working.

**Pull request merged**